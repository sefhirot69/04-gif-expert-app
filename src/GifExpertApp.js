import React, {useState} from "react";
import AddCategory from "./components/AddCategory";


const GifExpertApp = () => {

    const [categories, setCategories] =
        useState(['One Punch', 'Samurai X', 'Dragon Ball']);

    // const handleAdd = () => {
    //     setCategories([...categories, 'Chicho']);
    //     // setCategories(cats => [...cats, 'HunterXHunter']);
    // }

    return <>
        <h2>GifExpertApp</h2>
        <AddCategory/>
        <hr/>
        <ol>
            {
                categories.map(category => {
                    return <li key={category}>{category}</li>
                })
            }
        </ol>
    </>
};

export default GifExpertApp;