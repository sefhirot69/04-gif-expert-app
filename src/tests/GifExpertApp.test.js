import {shallow} from "enzyme";
import GifExpertApp from "../GifExpertApp";

describe('Prueba de GifExpertApp', () => {

    const wrapper = shallow(<GifExpertApp/>)

    test('debe mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot();
    });

    test('debe mostrar el componente correctamente con Goku como categoria', () => {

        const wrapper = shallow(<GifExpertApp defaultCategories={['Goku']}/>)
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(1);
        expect(wrapper.find('AddCategory').length).toBe(1);
    });
});