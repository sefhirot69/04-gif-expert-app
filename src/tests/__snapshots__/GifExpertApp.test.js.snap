// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de GifExpertApp debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;

exports[`Prueba de GifExpertApp debe mostrar el componente correctamente con Goku como categoria 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="Goku"
    key="Goku"
  />
</Fragment>
`;
